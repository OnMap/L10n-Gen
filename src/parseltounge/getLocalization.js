import axios from 'axios';
import get from 'lodash/get';

const { parseltongueHost } = require('../constants');

export async function uploadKeys(
  appId,
  token,
  host = parseltongueHost.prod
) {
  if (!appId || !token) {
    throw new Error('Missing params!');
  }

  try {
    const response = await axios.post(`${host}/v1/translations`, {
        app_id: appId,
    }, {
        headers: {
        access_token: token
        }
    });

    if (response && response.data) { return response.data; }

        throw new Error('Could not upload keys');
    }

    catch (err) {
        const data = get(err, 'response.data');
        if (data) {
            throw new Error(data && data.message);
        }     
    
        throw err;
    }

}


module.exports = {
    uploadKeys
}