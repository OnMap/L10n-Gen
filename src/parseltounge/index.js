import axios from 'axios';
import get from 'lodash/get';

const { parseltongueHost } = require('../constants');

export async function uploadKeys(keys = [], appId = '', token = '', host = parseltongueHost.prod) {
  //   if (!data.topic_id) {
  //     throw new BadRequestError('Topic is not defined. Check your app config');
  //   }

  try {
    const response = await axios.post(
      `${host}/v1/keys`,
      {
        app_id: appId,
        names: keys
      },
      {
        headers: {
          access_token: token
        }
      }
    );

    if (response && response.data) {
      // logger.info(response.data);
      return response.data;
    }

    throw new Error('Could not upload keys');
  } catch (err) {
    const data = get(err, 'response.data');
    throw new Error(data && data.message);
  }
}
