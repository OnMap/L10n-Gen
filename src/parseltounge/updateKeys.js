import axios from 'axios';
import get from 'lodash/get';

const { parseltongueHost } = require('../constants');

export async function uploadKeys(
  keys,
  appId,
  token,
  host = parseltongueHost.prod
) {
  if (!appId || !token || !keys) {
    throw new Error('Missing params!');
  }

  try {
    const response = await axios.post(`${host}/v1/keys`, {
      app_id: appId,
      names: keys
    }, {
      headers: {
        access_token: token
      }
    });

    if (response && response.data) {
      return response.data;
    }

    throw new Error('Could not upload keys');
  } catch (err) {
    const data = get(err, 'response.data');
    if (data) {
      throw new Error(data && data.message);
    }     
    
    throw err;
  }
}

module.exports = {
    uploadKeys
}
