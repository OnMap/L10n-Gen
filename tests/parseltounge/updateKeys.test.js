import axios from 'axios';
import { uploadKeys } from '../../src/parseltounge/updateKeys';

describe('Update Keys', () => {
  const testingHost = 'https://dev-parseltongue.onmap.co.il';
  const request = {
    token: 'token',
    appId: 'eca59a39-ca2e-4d22-80b3-18e0b5a8f741',
    keys: ['testKey1', 'testKey2']
  }

  it('upload all keys', async () => {
    axios.post = jest.fn(() => Promise.resolve({ data: 'OK' }));
    expect(uploadKeys(request.keys, request.appId, request.token, testingHost)).resolves.toMatchSnapshot();
    axios.post.mockRestore();
  });

  it('handles empty response', async () => {
    axios.post = jest.fn(() => Promise.resolve({ data: '' }));
    expect(uploadKeys(request.keys, request.appId, request.token, testingHost)).rejects.toMatchSnapshot();
    axios.post.mockRestore();
  });

  it('handles error on request', async () => {
    axios.post = jest.fn(() => Promise.reject({ response: { data: { message: 'not-ok' } } }));
    expect(uploadKeys(request.keys, request.appId, request.token, testingHost)).rejects.toMatchSnapshot();
    axios.post.mockRestore();
  });

});

