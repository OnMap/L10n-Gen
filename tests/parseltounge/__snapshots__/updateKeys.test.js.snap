// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Keys handles empty response 1`] = `[Error: Could not upload keys]`;

exports[`Update Keys handles error on request 1`] = `[Error: not-ok]`;

exports[`Update Keys upload all keys 1`] = `"OK"`;
